roles/example-cnf-catalog/tasks/main.yml

```diff
@@ -68,11 +68,16 @@
   until: pkg_manifest_trex.resources|length != 0
   failed_when: pkg_manifest_trex.resources|length == 0
 
-- name: create namespace
+- name: Create namespace
   k8s:
-    api_version: v1
-    name: "{{ cnf_namespace }}"
-    kind: Namespace
+    definition:
+      api_version: v1
+      kind: Namespace
+      metadata:
+        name: "{{ cnf_namespace }}"
+        labels:
```

Comment:
Likely won't affect other OCP versions but it would be best to only inject these labels where it matters, i.e. 4.12+

This can be done with a template a la:

```yaml
    vars:
      ns_template: |-
        apiVersion: v1
        kind: Namespace
        metadata:
          name: "{{ cnf_namespace }}"
          {% if ocp_version is version("4.12", ">=") -%}
          labels:
            required-labels-go-here
          {% endif -%}
    community.kubernetes.k8s:
      state: present
      definition: "{{ ns_template | from_yaml }}"
```

Reply:
we discussed with @betoredhat that probably that will do no harm for the previous OCP versions, I'll test

Reply:
@tonyskapunk I tested on 4.7, 4.10, 4.11, and 4.12 - these labels do not harm. So I prefer to keep them everywhere to avoid if/else code. We could discuss it tomorrow.